src/crate/client/sqlalchemy/compat/core10.py

```diff
@@ -0,0 +1,256 @@
+
+        set_clauses = []
+
+        for k, v in crud_params:
+            clause = k._compiler_dispatch(self,
+                                          include_table=include_table) + \
+                ' = ' + v
+            set_clauses.append(clause)
+
+        for k, v in update_stmt.parameters.items():
+            if isinstance(k, str) and '[' in k:
+                bindparam = sa.sql.bindparam(k, v)
+                set_clauses.append(k + ' = ' + self.process(bindparam))
+
+        text += ', '.join(set_clauses)
+        # [10] CrateDB patch end.
+
+        if self.returning or update_stmt._returning:
+            if not self.returning:
+                self.returning = update_stmt._returning
+            if self.returning_precedes_values:
+                text += " " + self.returning_clause(
+                    update_stmt, self.returning)
+
+        if extra_froms:
+            extra_from_text = self.update_from_clause(
+                update_stmt,
+                update_stmt.table,
+                extra_froms,
+                dialect_hints,
+                **kw)
+            if extra_from_text:
+                text += " " + extra_from_text
+
+        if update_stmt._whereclause is not None:
+            t = self.process(update_stmt._whereclause)
+            if t:
+                text += " WHERE " + t
+
+        limit_clause = self.update_limit_clause(update_stmt)
+        if limit_clause:
+            text += " " + limit_clause
+
+        if self.returning and not self.returning_precedes_values:
+            text += " " + self.returning_clause(
+                update_stmt, self.returning)
+
+        return text
+
+
+def _get_crud_params(compiler, stmt, **kw):
+    """create a set of tuples representing column/string pairs for use
+    in an INSERT or UPDATE statement.
+
+    Also generates the Compiled object's postfetch, prefetch, and
+    returning column collections, used for default handling and ultimately
+    populating the ResultProxy's prefetch_cols() and postfetch_cols()
+    collections.
+
+    """
+
+    compiler.postfetch = []
+    compiler.insert_prefetch = []
+    compiler.update_prefetch = []
+    compiler.returning = []
+
+    # no parameters in the statement, no parameters in the
+    # compiled params - return binds for all columns
+    if compiler.column_keys is None and stmt.parameters is None:
+        return [
+            (c, _create_bind_param(compiler, c, None, required=True))
+            for c in stmt.table.columns
+        ]
+
+    if stmt._has_multi_parameters:
+        stmt_parameters = stmt.parameters[0]
+    else:
+        stmt_parameters = stmt.parameters
+
+    # getters - these are normally just column.key,
+    # but in the case of mysql multi-table update, the rules for
+    # .key must conditionally take tablename into account
+    (
+        _column_as_key,
+        _getattr_col_key,
+        _col_bind_name,
+    ) = _key_getters_for_crud_column(compiler, stmt)
```

Comment:
CodeQL is comparing apples with oranges here and below. I think it does not get SQLAlchemy's versions right, and compares the wrong versions.

Reply:
Let's see if it will get fixed by GH-499.

Reply:
I think CodeQL will not be smart enough to handle this case, when a code base contains code which calls functions or methods in 3rd-party libraries with different signatures, because of their evolutions across different versions.

So, I think we will have to go with manually dismissing those admonitions. Please correct me if I am wrong.